### ISSUE TITLE
only FLOAT is supported for tensorflow nodes....???

### ISSUE THREAD

**Issue by vinayonchip** [ARM-software/armnn#82]
I have trained a custom tensorflow graph and when i run it for inference on a raspberry pi3, I get the following exception.

terminate called after throwing an instance of 'armnn::ParseException'
  what():  Currently only FLOAT is supported for tensorflow nodes (apart from Const)
Aborted

**Comment by MatthewARM** [comment]
That's interesting @vinayonchip, do you know what data types are in your graph?

**Comment by vinayonchip** [comment]
I am using integers as well as floats.

**Comment by MatthewARM** [comment]
Hi @vinayonchip ArmNN doesn't support integer tensors, except for the QASYMM8 quantized type. If you could describe a bit more about your use-case, we could maybe find a workaround, or add your request to our roadmap?

**Comment by MatthewARM** [comment]
I've had a bit of a think about this, and if the tf.gather() operation is the last stage in your network, maybe you could just run the rest of the net in ArmNN and roll your own word lookups as the final stage?

If you fundamentaily need an integer loss function for your application then that's going to be a bigger problem, but if you can get the net to work with float except for the word embeddings, you should be able to use ArmNN for the difficult bits.

Whatever you decide, I've noted down your interest in int32 support and tf.gather() for our roadmap.

**Comment by MatthewARM** [comment]
Hi @vinayonchip we're looking at requirements for language processing models at the moment. Would it be possible for you to share more details about your model? Is it an open-source architecture or something proprietary?

**Comment by vinayonchip** [comment]
> Hi @vinayonchip we're looking at requirements for language processing models at the moment. Would it be possible for you to share more details about your model? Is it an open-source architecture or something proprietary?

Hi @MatthewARM ,
We are using Google's Transformer model for our language processing task. Hence we have to use both the types(INTS, FLOATS) :
Ints for inputs and outputs.
Floats for loss function/weights

https://github.com/tensorflow/models/blob/master/official/transformer/model/transformer.py

Thanks in advance.!

**Comment by MatthewARM** [comment]
Thanks @vinayonchip that's very useful. We're keen to support Transformer but we haven't yet figured out all of the bits of work that are necessary.

**Comment by MatthewARM** [comment]
Closing this as the Tensorflow parser now supports Signed32 tensors and the Gather operator. Movement in the right direction although the full Transformer net will likely require other things too.

**Comment by shadabshamim786** [comment]
Can I see the commits made to support INT32 for tensor flow nodes

**Comment by MatthewARM** [comment]
db2b160bf9e7759d0157dfa57ee940290f5170e3
